src/main/scala/io/iohk/ethereum/vm/Stack.scala

```diff
@@ -0,0 +1,95 @@
+package io.iohk.ethereum.vm
+
+import io.iohk.ethereum.utils.EitherExtensions._
+
+object Stack {
+  /**
+    * Stack max size as defined in the YP (9.1)
+    */
+  val DefaultMaxSize = 1024
+
+  def empty(maxSize: Int = DefaultMaxSize): Stack =
+    new Stack(Vector(), maxSize)
+}
+
+//TODO: consider a List with head being top of the stack (DUP,SWAP go at most the depth of 16)
+/**
+  * Stack for the EVM. Instruction pop their arguments from it and push their results to it.
+  * The Stack handles overflow and underflow errors.
+  */
+class Stack private(private val underlying: Vector[DataWord], val maxSize: Int) {
```

Comment:
It's a matter of taste. I didn't have a strong opinion; Radek chose ~case~. Let's keep it as it is if you don't mind.